client/src/components/GraphViewerComponent/GraphViewerComponent.js

```diff
@@ -177,7 +177,7 @@ class GraphViewerComponent extends React.Component {
         this.cyRef.current.clearSelection();
       }
       const allTwins = await this.getTwinsData(query, overlayResults);
-      if (!this.state.couldNotDisplay) {
+      if (allTwins.length > 1) {
```

Comment:
Should this be greater than zero?

I think you can create a self-referential relationship?